src/deploy/functions/tasks.ts

```diff
@@ -0,0 +1,279 @@
+import * as clc from "cli-color";
+
+import * as logger from "../../logger";
+import * as utils from "../../utils";
+import { CloudFunctionTrigger } from "./deploymentPlanner";
+import { cloudfunctions, cloudscheduler } from "../../gcp";
+import * as deploymentTool from "../../deploymentTool";
+import * as helper from "../../functionsDeployHelper";
+import { RegionalDeployment } from "./deploymentPlanner";
+import { OperationPollerOptions, pollOperation } from "../../operation-poller";
+import { functionsOrigin } from "../../api";
+import Queue from "../../throttler/queue";
+import { getHumanFriendlyRuntimeName } from "../../parseRuntimeAndValidateSDK";
+import { deleteTopic } from "../../gcp/pubsub";
+import { DeploymentTimer } from "./deploymentTimer";
+import { ErrorHandler } from "./errorHandler";
+
+// TODO: Tune this for better performance.
+const defaultPollerOptions = {
+  apiOrigin: functionsOrigin,
+  apiVersion: cloudfunctions.API_VERSION,
+  masterTimeout: 150000,
```

Comment:
nit* can we write this as

```
40 * 60 * 60 // 40 hours
```

Also - did I get this right? 40 hours looks like a very generous timeout.

Reply:
This is in ms, so its 150 seconds. I'll add a comment clarifying this for now. I'd like to make some improvements to throttler at some point in the future since it wasn't the clearest to work with here - when I do so, I'll rename this option to masterTimeoutMs for clarity.